test/unit/validator/proposer_test.exs

```diff
@@ -29,7 +31,11 @@ defmodule Unit.Validator.ProposerTests do
       <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 64>>
 
-    {:ok, signed_block} = Proposer.construct_block(pre_state, privkey)
+    validator_index = 63
+
+    {:ok, state} = StateTransition.process_slots(pre_state, 1)
+
+    {:ok, signed_block} = Proposer.construct_block(state, validator_index, privkey)
```

Comment:
let's call this `state` pre. Does it verify that `state_transition(pre, block)` is valid?

Reply:
We might want to add that either to the test or to the `construct_block` function.

Reply:
Done